Redo snap redis mon

david-deboer opened the pull request:
This updates hera_corr_cm to use just the one status:snap key in redis (so converts in ingest, not in the hera_corr_f daemon output).  It also plumbs through the remaining snap status keys.  This is not the new array format, but the incremental next step in the process.

bhazelton commented on hera_corr_cm/hera_corr_cm.py:
```diff
@@ -253,36 +253,54 @@ def get_f_status(self):
+            dest_is_configured (bool): True if dest_is_configured
+            version (str)      : Version of firmware installed
+            sample_rate (float): Sample rate in MHz
+            input (str)        : comma-delimited list of stream inputs, e.g. adc,adc,adc,adc,adc,adc
```
what does this look like for digital noise?

david-deboer replied:
Found and added.

bhazelton replied:
thanks!

bhazelton commented on hera_corr_cm/hera_corr_cm.py:
```diff
@@ -253,36 +253,54 @@ def get_f_status(self):
-                    rv[host][key] = "None"
-        return rv
+                    f_status[host][key] = cfunc(stats[host][ckey])
+                except:  # noqa
```
bare excepts aren't great...

david-deboer replied:
I don't know enough to specify exceptions -- I'll make an issue.

bhazelton commented on hera_corr_cm/hera_corr_cm.py:
```diff
@@ -197,12 +197,12 @@ def get_eq_coeffs(self, ant, pol):
         try:
             t = float(v['time'])
-        except:
+        except:  # noqa
```
bare excepts aren't great. Can we fix it instead of ignoring the warnings about it?

david-deboer replied:
I'll make an issue and take out the noqas

bhazelton commented on hera_corr_cm/hera_corr_cm.py:
```diff
@@ -197,12 +197,12 @@ def get_eq_coeffs(self, ant, pol):
         try:
             coeffs = np.array(json.loads(v['values']), dtype=np.float)
-        except:
+        except:  # noqa
```
bare excepts aren't great.

david-deboer replied:
ditto

bhazelton commented on hera_corr_cm/hera_corr_cm.py:
```diff
@@ -427,7 +474,7 @@ def get_hashpipe_status(self):
                     if np.isnan(fields[k]):
                         continue
-                except:
+                except:  # noqa
```
bare excepts aren't great.

david-deboer replied:
I'll remove noqa and add issue per above.

david-deboer commented on hera_corr_cm/hera_corr_cm.py:
```diff
@@ -253,36 +253,54 @@ def get_f_status(self):
         Keys of returned dictionaries are snap hostnames. Values of this dictionary are
         status key/val pairs.
 
+        For the conv_info dictionary, the format is:
+            key: name of the variable in the return dictionary from this method
+            tuple:  (redis key name, conversion method from redis to this method).
```
Done.

david-deboer commented on hera_corr_cm/hera_corr_cm.py:
```diff
@@ -291,6 +309,10 @@ def get_ant_status(self):
         Keys of returned dictionaries are of the form "<antenna number>:"<e|n>". Values of
         this dictionary are status key/val pairs.
 
+        For the conv_info dictionary, the format is:
+            key: name of the variable in the return dictionary from this method
+            tuple:  (redis key name, conversion method from redis to this method).
```
Done.

david-deboer commented on hera_corr_cm/hera_corr_cm.py:
```diff
@@ -321,76 +343,101 @@ def get_ant_status(self):
         Keys of returned dictionaries are of the form "<SNAP hostname>:"<SNAP input number>".
         Values of this dictionary are status key/val pairs.
 
+        For the conv_info dictionary, the format is:
+            key: name of the variable in the return dictionary from this method
+            tuple:  (redis key name, conversion method from redis to this method).
```
Done.

david-deboer commented on hera_corr_cm/hera_corr_cm.py:
```diff
@@ -321,76 +343,101 @@ def get_ant_status(self):
         These keys are:
-            eq_coeffs (list of floats) : Digital EQ coefficients for this antenna
-            histogram (list of ints) : Two-dim list: [[bin_centers][counts]] represent ADC histogram
+            eq_coeffs (list of floats) : Digital EQ coefficients for this host
+            histogram (list of ints) : List of counts representing ADC histogram
             autocorrelation (list of floats) : Autocorrelation spectrum
             timestamp (datetime) : Asynchronous timestamp that these status entries were gathered
-
+            mean (float): mean of power for this host:stream
+            rms (float): rms of power for this host:stream
+            power (float): total power for this host:stream
```
This is currently done this way for backward compatibility.  The information is duplicated in get_snaprf_status and the difference is how they are keyed in redis.  One is done with looping over ant/pol, the other is done looping over snap hostname, so the code can't be combined.

bhazelton replied:
I'm confused. Are you saying that the information being returned from these two functions are from separate redis keys, which just happen to have the same information?

bhazelton commented on hera_corr_cm/hera_corr_cm.py:
```diff
@@ -320,77 +345,115 @@ def get_ant_status(self):
+        Returns
+        -------
+        dict
+            keys are the parameter names to be stored, values are the parameter values
```
I'd move the description of the returned dict under here. What is the second sentence in this docstring talking about? Is it about the output dict or is it about the stuff in redis?

bhazelton commented on hera_corr_cm/hera_corr_cm.py:
```diff
@@ -320,77 +345,115 @@ def get_ant_status(self):
         """
         Return a dictionary of SNAP input stats.
 
         Keys of returned dictionaries are of the form "<SNAP hostname>:"<SNAP input number>".
         Values of this dictionary are status key/val pairs.
```
Is this for the dict returned by this function? Or something else?

bhazelton commented on hera_corr_cm/hera_corr_cm.py:
```diff
@@ -320,77 +345,115 @@ def get_ant_status(self):
         These keys are:
-            eq_coeffs (list of floats) : Digital EQ coefficients for this antenna
-            histogram (list of ints) : Two-dim list: [[bin_centers][counts]] represent ADC histogram
+            eq_coeffs (list of floats) : Digital EQ coefficients for this host
+            histogram (list of ints) : List of counts representing ADC histogram
             autocorrelation (list of floats) : Autocorrelation spectrum
             timestamp (datetime) : Asynchronous timestamp that these status entries were gathered
-
+            mean (float): mean of power for this host:stream
+            rms (float): rms of power for this host:stream
+            power (float): total power for this host:stream
```
If these describe the dict returned by this function, I'd put them under the Returns section.

david-deboer commented:
I've fixed/updated the Return docstring and removed the duplicated content earlier in the docstring.  ant_status and snaprf_status contain duplicated content but keyed differently (one on hostname:stream one on ant:pol).  This is to keep compatibility with the existing code called from hera_mc.  I'm assuming it gets duplicated in hera_mc as well.

bhazelton commented:
As far as I can tell, `hera_mc` doesn't call `get_snaprf_status`, it only calls `get_f_status`, so we don't need to keep `get_rf_status` around for that package. Maybe something else needs it though. We should find out eventually.

I now understand why you need two functions (in that what they return has a different structure). But if they are reading the same redis keys then there is some duplication that could be removed. But if we're just going to get rid of `get_rf_status` maybe that doesn't matter. If that's the plan, we should make a note in the docstring that we're planning to get rid of it.

bhazelton reviewed (approved):
Looks good.
